pkg/collector/corechecks/snmp/fetch.go

```diff
@@ -0,0 +1,136 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"sort"
+)
+
+func fetchScalarOidsOne(session sessionAPI, oids []string) (map[string]snmpValue, error) {
+	// Get results
+	log.Debugf("fetchScalarOidsByBatch() oids: %v", oids)
+	results, err := session.Get(oids)
+	log.Debugf("fetchColumnOids() results: %v", results)
+	if err != nil {
+		return nil, err
+	}
+	return resultToScalarValues(results), nil
+}
+
+func fetchScalarOidsByBatch(session sessionAPI, oids []string, oidBatchSize int) (map[string]snmpValue, error) {
+	// Get results
+	// TODO: Improve batching algorithm and make it more readable
+	retValues := make(map[string]snmpValue)
+
+	if oidBatchSize == 0 {
+		// TODO: test me
+		return nil, fmt.Errorf("oidBatchSize cannot be 0")
+	}
+
+	// Make matches
+	for i := 0; i < len(oids); i += oidBatchSize {
+		j := i + oidBatchSize
+		if j > len(oids) {
+			j = len(oids)
+		}
+		fetchOids := oids[i:j]
+
+		results, err := fetchScalarOidsOne(session, fetchOids)
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range results {
+			retValues[k] = v
+		}
+	}
+	return retValues, nil
+}
+
+func fetchColumnOids(session sessionAPI, oids map[string]string, oidBatchSize int) (map[string]map[string]snmpValue, error) {
+	// Get results
+	// TODO: Improve batching algorithm and make it more readable
+	//   -
+
+	if oidBatchSize == 0 {
+		// TODO: test me
+		return nil, fmt.Errorf("oidBatchSize cannot be 0")
+	}
+
+	retValues := make(map[string]map[string]snmpValue)
+	columnOids := make([]string, 0, len(oids))
+
+	for k := range oids {
+		columnOids = append(columnOids, k)
+	}
+
+	// Make matches
+	for i := 0; i < len(oids); i += oidBatchSize {
+		j := i + oidBatchSize
+		if j > len(oids) {
+			j = len(oids)
+		}
+		batchColumnOids := columnOids[i:j]
+		fetchOids := make(map[string]string)
+		for _, oid := range batchColumnOids {
+			fetchOids[oid] = oids[oid]
+		}
+
+		results, err := fetchColumnOidsOne(session, fetchOids)
+		if err != nil {
+			return nil, err
+		}
+		for columnOid, instanceOids := range results {
+			if _, ok := retValues[columnOid]; ok {
+				for oid, value := range instanceOids {
+					retValues[columnOid][oid] = value
+				}
+			} else {
+				retValues[columnOid] = instanceOids
+			}
+		}
+	}
+	return retValues, nil
+}
+
+func fetchColumnOidsOne(session sessionAPI, oids map[string]string) (map[string]map[string]snmpValue, error) {
```

Comment:
what is the `One` standing for ?

Reply:
Sure, it's for one batch. Renaming to `fetchColumnOidsOneBatch`